core/src/main/java/com/sequenceiq/cloudbreak/controller/validation/template/TemplateValidator.java

```diff
@@ -174,11 +186,17 @@ private void validateVolumeCount(VolumeTemplate value, VmType vmType, VolumePara
     }
 
     private void validateVolumeSize(VolumeTemplate value, VmType vmType, VolumeParameterType volumeParameterType,
-        ValidationResult.ValidationResultBuilder validationBuilder) {
+        ValidationResult.ValidationResultBuilder validationBuilder, InstanceGroup instanceGroup) {
         if (vmType != null && needToCheckVolume(volumeParameterType, value.getVolumeCount())) {
             VolumeParameterConfig config = vmType.getVolumeParameterbyVolumeParameterType(volumeParameterType);
             if (config != null) {
-                if (value.getVolumeSize() > config.maximumSize()) {
+                if (isIDBrokerInstanceGroup(instanceGroup)) {
+                    if (value.getVolumeSize() > config.maximumSize()) {
+                        // IDBroker does not use data volume, so its volume size should be zero
+                        // To be backward-compatible, we only check max limit and allow the min limit to be zero for IDBroker
+                        validationBuilder.error(String.format("Max allowed volume size for '%s': %s", vmType.value(), config.maximumSize()));
+                    }
+                } else if (value.getVolumeSize() > config.maximumSize()) {
                     validationBuilder.error(String.format("Max allowed volume size for '%s': %s", vmType.value(), config.maximumSize()));
                 } else if (value.getVolumeSize() < config.minimumSize()) {
```

Comment:
Insert `!isIDBrokerInstanceGroup(instanceGroup) && ` at the beginning of condition.